### PR TITLE
Improve solution and project settings for dev extensions

### PR DESCRIPTION
**Pull request by boscohyun**
This pull request improve Lib9c solution and projects.

- Add `Debug DevEx` configuration based on `Debug`.
- `Lib9c` project include `Lib9c.DevExtensions` project when `Debug DevEx`.

### CODE REVIEW

**Comment by longfin** [comment]
> Add Debug DevEx configuration based on Debug.

IMHO, `Debug DevEx` configuration seems a little bit verbose and overlaps with `Debug`. What are some reasons to separate `Debug` and `Debug DevEx`?

File: .Lib9c.Tools/Lib9c.Tools.csproj [hunk 1]
```diff
@@ -4,20 +4,30 @@
...
         <TargetFramework>net6.0</TargetFramework>
         <AssemblyName>9c-tools</AssemblyName>
+        <Configurations>Debug;Release;Debug DevEx</Configurations>
+        <Platforms>AnyCPU</Platforms>
     </PropertyGroup>
 
-    <PropertyGroup Condition="
-      '$([System.Runtime.InteropServices.RuntimeInformation]::
-        OSArchitecture.ToString())' == 'Arm64' ">
+    <PropertyGroup Condition="&#xA;      '$([System.Runtime.InteropServices.RuntimeInformation]::&#xA;        OSArchitecture.ToString())' == 'Arm64' ">
```
Comment (moreal) [inline .Lib9c.Tools/Lib9c.Tools.csproj:11]: It seems to have useless characters `&#xA;`

**Review by moreal — APPROVED** [review]

**Comment by boscohyun** [comment]
> > Add Debug DevEx configuration based on Debug.
> 
> IMHO, `Debug DevEx` configuration seems a little bit verbose and overlaps with `Debug`. What are some reasons to separate `Debug` and `Debug DevEx`?

@longfin First of all, It was needed to ref the `Lib9c.DevExtensions project` from the `Lib9c` project to use some codes which declared in the `Lib9c.DevExtensions` project. For example, To use the `class ActionEx : GameAction` which is declared in the `Lib9c.DevExtensions` project, as an `PolymorphicAction<T>` in `Lib9c` assembly.

So I choose the way that seperate the configurations of solution and project.

I usually develop with `Debug` configuration in IDE. And I can expect that the `Lib9c` assembly does not contains the `Lib9c.DevExtensions` project(or codes) when if the configuration is just `Debug`.
And I intended a devs can expect the `Lib9c` assembly contains the `Lib9c.DevExtensinos` project(or codes) when set the configuration to `Debug DevEx`.

If the name of `Debug DevEx` is too verbose, it can be `DevEx`.
Thanks!

**Comment by longfin** [comment]
Thanks for explaining, `DevEx` seems fine.

**Review by longfin — APPROVED** [review]

File: .Lib9c.Benchmarks/Lib9c.Benchmarks.csproj [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
     <ItemGroup>
       <PackageReference Include="Serilog" Version="2.10.0" />
-      <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
+      <PackageReference Include="Serilog.Sinks.Console" Version="4.0.1" />
```
[the remark this answers is not in the document]
Comment (boscohyun) [reply .Lib9c.Benchmarks/Lib9c.Benchmarks.csproj]: This change was required when build with `DevEx` configuration.

```
  Lib9c.Benchmarks.csproj: [NU1605] Detected package downgrade: Serilog.Sinks.Console from 4.0.1 to 3.1.1. Reference the package directly from the project to select a different version. 
 Lib9c.Benchmarks -> Lib9c -> Serilog.Sinks.Console (>= 4.0.1) 
 Lib9c.Benchmarks -> Serilog.Sinks.Console (>= 3.1.1)
```
`Lib9c.DevExtensions` and other project are use the package 4.0.1. And the `Lib9c` project contains the `Lib9c.DevExtensions` project when the configuration is `DevEx`.

image

**Pull request merged**